examples/tutorials/api/priors.py

```diff
@@ -0,0 +1,419 @@
+# The value of the prior depends on the value of the index. If the index
+# value equals the Gaussians mean (here 2.1), the prior is zero. This
+# means that nothing is added to the cash statistics, and this value is
+# favoured in the fit. If the value of the index deviates from the mean of
+# 2.1, a prior value > 0 is added to the cash statistics.
+#
+
+prior_stat_sums = []
+with model_prior.parameters.restore_status():
+    i_scan = np.linspace(1.8, 2.3, 100)
+    for a in i_scan:
+        model_prior.parameters["index"].value = a
+        prior_stat_sums.append(model_prior.parameters.prior_stat_sum())
+
+plt.plot(
+    i_scan,
+    prior_stat_sums,
+    color="tab:orange",
+    linestyle="dashed",
+    label=f"Gaussian Prior: \n$\mu = {gaussianprior.mu.value}$; $\sigma = {gaussianprior.sigma.value}$",
+)
+plt.axvline(x=gaussianprior.mu.value, color="red")
+
+plt.xlabel("Index Value")
+plt.ylabel("Prior")
+plt.legend()
+plt.xlim(2.0, 2.2)
+plt.ylim(-0.05, 1.1)
+plt.show()
+
+
+######################################################################
+# Fitting an Asimov Dataset with and without the Prior
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# Now, we copy the dataset consisting of the power-law source and assign
+# the model with the Gaussian prior set on its index to it. Both of the
+# datasets are fitted.
+#
+
+dataset1_prior = dataset1.copy()
+dataset1_prior.models = model_prior.copy(name="prior-model")
+
+fit = Fit()
+results = fit.run(dataset1)
+
+fit_prior = Fit()
+results_prior = fit_prior.run(dataset1_prior)
+
+
+######################################################################
+# The Likelihood profiles can be computed for both the datasets. Hereby,
+# the likelihood gets computed for different values of the index. For each
+# step the other free parameters are getting reoptimized.
+#
+
+dataset1.models.parameters["index"].scan_n_values = 20
+dataset1_prior.models.parameters["index"].scan_n_values = 20
+
+scan = fit.stat_profile(datasets=dataset1, parameter="index", reoptimize=True)
+
+scan_prior = fit_prior.stat_profile(
+    datasets=dataset1_prior, parameter="index", reoptimize=True
+)
+
+
+######################################################################
+# Now, we can compare the two Likelihood scans. In the first case, we did
+# not set any prior. This means that only the cash statistic itself is
+# getting minimized. The cash statistics minimum is the actual value of
+# the index we used for the simulation (:math:`2.3`). Therefore, the
+# best-fit value was found to be :math:`2.3`. Note how the error bars
+# correspond to the :math:`1\sigma` error, i.e. where the stat sum equals
+# the minimum + 1.
+#
+# The plot also shows the prior we set on the index for the second
+# dataset. The scan was computed above. If the prior is added to the cash
+# statistics, one ends up with the posterior function. The posterior is
+# minimized during the second fit. Its minimum is between the priors and
```

Comment:
If thing here you could explicitly use the two terminologies
either penalized maximum likelihood or the maximum a posteriori estimation